services/supervisor/tests/supervisor/config.rs

```diff
@@ -0,0 +1,740 @@
+
+    let confirm = ConfigVote {
+        propose_hash: proposal_hash,
+    };
+    // Sign and send confirmation transaction by second validator
+    let keys = testkit.network().validators()[1].service_keypair();
+    testkit
+        .create_block_with_transaction(confirm.clone().sign(
+            SUPERVISOR_INSTANCE_ID,
+            keys.0,
+            &keys.1,
+        ))
+        .transactions[0]
+        .status()
+        .expect("Transaction with confirmations discarded.");
+
+    // Sign confirmation transaction by third validator
+    let keys = testkit.network().validators()[2].service_keypair();
+    testkit
+        .create_block_with_transaction(confirm.sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1))
+        .transactions[0]
+        .status()
+        .expect("Transaction with confirmation discarded.");
+
+    assert_eq!(config_propose_entry(&testkit), None);
+    assert_eq!(testkit.consensus_config(), consensus_config);
+}
+
+#[test]
+fn test_send_confirmation_by_initiator() {
+    let mut testkit = testkit_with_supervisor(4);
+    let initiator_id = testkit.network().us().validator_id().unwrap();
+
+    let consensus_config = consensus_config_propose_first_variant(&testkit);
+    let config_proposal = ConfigProposeBuilder::new(CFG_CHANGE_HEIGHT)
+        .extend_consensus_config_propose(consensus_config.clone())
+        .config_propose();
+    let proposal_hash = config_proposal.object_hash();
+
+    testkit
+        .create_block_with_transaction(sign_config_propose_transaction(
+            &testkit,
+            config_proposal,
+            initiator_id,
+        ))
+        .transactions[0]
+        .status()
+        .expect("Transaction with change propose discarded.");
+
+    // Try to send confirmation transaction by the initiator
+    let keys = testkit.network().us().service_keypair();
+    let signed_confirm = ConfigVote {
+        propose_hash: proposal_hash,
+    }
+    .sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1);
+    let err = testkit
+        .create_block_with_transaction(signed_confirm)
+        .transactions[0]
+        .status()
+        .unwrap_err()
+        .clone();
+    assert_eq!(err.kind, Error::AttemptToVoteTwice.into());
+}
+
+#[test]
+fn test_propose_config_change_by_incorrect_validator() {
+    let mut testkit = testkit_with_supervisor(1);
+
+    let consensus_config = consensus_config_propose_first_variant(&testkit);
+    let config_proposal = ConfigProposeBuilder::new(CFG_CHANGE_HEIGHT)
+        .extend_consensus_config_propose(consensus_config.clone())
+        .config_propose();
+
+    let keys = crypto::gen_keypair();
+    let signed_confirm = config_proposal.sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1);
+
+    let err = testkit
+        .create_block_with_transaction(signed_confirm)
+        .transactions[0]
+        .status()
+        .unwrap_err()
+        .clone();
+    assert_eq!(err.kind, Error::UnknownAuthor.into());
```

Comment:
The same here.

Reply:
Link to the comment to which "the same" is addressed: [link](https://github.com/exonum/exonum/pull/1484#discussion_r331980809)

(it may be not clear if look on the main PR page)